pycalphad/core/equilibrium.py

```diff
@@ -239,22 +247,25 @@ def equilibrium(dbf, comps, phases, conditions, output=None, model=None,
     if verbose:
         print('[done]', end='\n')
 
-    # 'calculate' accepts conditions through its keyword arguments
-    grid_opts = calc_opts.copy()
-    statevar_strings = [str(x) for x in state_variables]
-    grid_opts.update({key: value for key, value in str_conds.items() if key in statevar_strings})
-    if 'pdens' not in grid_opts:
-        grid_opts['pdens'] = 500
-    grid = calculate(dbf, comps, active_phases, model=models, fake_points=True,
-                     callables=callables, output='GM', parameters=parameters, 
-                     to_xarray=False, **grid_opts)
+    if global_min:
+        # 'calculate' accepts conditions through its keyword arguments
+        grid_opts = calc_opts.copy()
+        statevar_strings = [str(x) for x in state_variables]
+        grid_opts.update({key: value for key, value in str_conds.items() if key in statevar_strings})
+        if 'pdens' not in grid_opts:
+            grid_opts['pdens'] = 500
+        grid = calculate(dbf, comps, active_phases, model=models, fake_points=True,
+                         callables=callables, output='GM', parameters=parameters,
+                         to_xarray=False, **grid_opts)
+    else:
+        grid = None
     coord_dict = str_conds.copy()
     coord_dict['vertex'] = np.arange(len(pure_elements) + 1)  # +1 is to accommodate the degenerate degree of freedom at the invariant reactions
     coord_dict['component'] = pure_elements
-    properties = starting_point(conds, state_variables, phase_records, grid)
+    properties = starting_point(conds, state_variables, phase_records, grid, given_starting_point=user_starting_point)
```

Comment:
IMO `starting_point` should be two different functions that are called separately here to emphasize to the reader that a `grid` is not needed for a `user_starting_point`. E.g. signatures:

`starting_point(conditions, state_variables, phase_records, grid)`
`custom_starting_point(conditions, state_variables, phase_records, compsets)`, where `compsets` is the `(phase, dof)` tuple.